### PR TITLE
EKF: fixed issue with scheduler init order in the EKF for replay

### PR DESCRIPTION
**Pull request by tridge**
Don't init till the scheduler has a loop rate
This fixes a FPE in replay

### CODE REVIEW

**Comment by peterbarker** [comment]
What's the path taken that allows the scheduler to return 0 from `get_loop_rate_hz`?

I'm concerned we might be papering over a bug which might bite us somewhere else.

AFAICS the parameter has to be zero - but it has a default non-zero value, so even before parameter load it should be good.

File: libraries/AP_NavEKF2/AP_NavEKF2.cpp [hunk 1]
```diff
@@ -634,7 +634,11 @@ bool NavEKF2::InitialiseFilter(void)
     imuSampleTime_us = AP::dal().micros64();
 
     // remember expected frame time
-    _frameTimeUsec = 1e6 / ins.get_loop_rate_hz();
+    const float loop_rate = ins.get_loop_rate_hz();
+    if (!is_positive(loop_rate)) {
+        return false;
+    }
```
Comment (magicrub) [inline libraries/AP_NavEKF2/AP_NavEKF2.cpp:640]: Should this instead set the param to it's default?

**Comment by tridge** [comment]
> What's the path taken that allows the scheduler to return 0 from `get_loop_rate_hz`?

I have the log (from a beaglebone blue) that reproduces this with replay:
http://uav.tridgell.net/tmp/blue-replay-error.bin

**Comment by tridge** [comment]
it turns out the problem is a missing RISI message at the start of the log. That means the DAL had not initialised the loop rate. I still think this is a reasonable fix though, as we don't want to init the EKF until we have that information

**Review by peterbarker — APPROVED** [review]
> it turns out the problem is a missing RISI message at the start of the log. That means the DAL had not initialised the loop rate. I still think this is a reasonable fix though, as we don't want to init the EKF until we have that information

I agree.  It's a "can't happen" and probably deserves a flow-of-control error.

Should note we can only afford so much sanity checking, 'though :-)

**Comment by tridge** [comment]
> I agree. It's a "can't happen" and probably deserves a flow-of-control error.

it's not flow of control error, it's a missing msg in a replay log

**Pull request merged**